packages/teuchos/core/src/Teuchos_RCPNode.hpp

```diff
@@ -162,24 +164,72 @@ class TEUCHOSCORE_LIB_DLL_EXPORT RCPNode {
   /** \brief . */
   int strong_count() const
     {
-      return count_[RCP_STRONG];
```

Comment:
The advantage of using a two-element `count_[]` array is that you can replace an if statement with a simple array index.  Performance profiling work showed this made a difference.